### PR TITLE
BTAT-2959: Updated vat summary to show number of due payments

### PR DESCRIPTION
**Pull request by davidmthornton**

### CODE REVIEW

File: app/controllers/VatDetailsController.scala [hunk 1]
```diff
@@ -22,10 +22,9 @@ import audit.AuditingService
 import audit.models.{ViewNextOpenVatObligationAuditModel, ViewNextOutstandingVatPaymentAuditModel}
 import config.AppConfig
 import javax.inject.{Inject, Singleton}
-
 import models.{ServiceResponse, User, VatDetailsDataModel}
 import models.obligations.{Obligation, VatReturnObligation, VatReturnObligations}
-import models.payments.Payment
+import models.payments.{Payment, Payments}
```
Comment (TomDudley92) [inline app/controllers/VatDetailsController.scala]: Can you remove the unused imports in this file please

File: app/services/VatDetailsService.scala [hunk 2]
```diff
@@ -21,12 +21,11 @@ import java.time.LocalDate
 import config.AppConfig
 import connectors.{FinancialDataConnector, VatApiConnector, VatSubscriptionConnector}
 import javax.inject.{Inject, Singleton}
-
 import models._
 import models.errors.{NextPaymentError, ObligationsError}
 import models.obligations.Obligation.Status._
-import models.obligations.{Obligation, VatReturnObligations}
-import models.payments.Payment
+import models.obligations.{VatReturnObligations}
```
Comment (TomDudley92) [inline app/services/VatDetailsService.scala]: You can remove the braces around `VatReturnObligations`

File: app/services/VatDetailsService.scala [hunk 3]
```diff
@@ -21,12 +21,11 @@ import java.time.LocalDate
...
+import models.payments.{Payments}
```
Comment (TomDudley92) [inline app/services/VatDetailsService.scala]: You can remove the braces around `Payments`

File: test/controllers/VatDetailsControllerSpec.scala [hunk 4]
```diff
@@ -39,13 +39,13 @@ import scala.concurrent.{ExecutionContext, Future}
 class VatDetailsControllerSpec extends ControllerBaseSpec {
 
   private trait DetailsTest {
-    val payment = Payment(
+    val payment = Payments(Seq(Payment(
```
Comment (TomDudley92) [inline test/controllers/VatDetailsControllerSpec.scala]: Can this val be renamed to `payments` as it is multiple payment models

File: test/services/VatDetailsServiceSpec.scala [hunk 5]
```diff
@@ -41,13 +41,13 @@ class VatDetailsServiceSpec extends ControllerBaseSpec {
       "#001"
     )
 
-    val payment: Payment = Payment(
+    val payments: Payments = Payments(Seq(Payment(
```
Comment (TomDudley92) [inline test/services/VatDetailsServiceSpec.scala]: Can more payment models be added to this sequence. With only one payment model being returned it doesn't truly test the ordering when it is being used in tests below

File: test/views/templates/NextPaymentSectionTemplateSpec.scala [hunk 6]
```diff
@@ -35,9 +35,7 @@ class NextPaymentSectionTemplateSpec extends ViewBaseSpec {
 
     "there is a payment to display" should {
 
-      val paymentDueDate = LocalDate.parse("2017-03-08")
-
-      lazy val view = views.html.templates.nextPaymentSection(Some(paymentDueDate), isOverdue = false, isError = false)
+      lazy val view = views.html.templates.nextPaymentSection(Some("2017-03-08"), hasMultiple = false, isOverdue = false, isError = false)
```
Comment (TomDudley92) [inline test/views/templates/NextPaymentSectionTemplateSpec.scala]: Can you remove the unused import in this file please

**Comment by davidmthornton** [comment]
I've made the changes as requested.

File: app/audit/models/ViewNextOutstandingVatPaymentAuditModel.scala [hunk 7]
```diff
@@ -17,17 +17,20 @@
...
+import models.payments.Payments
 
 case class ViewNextOutstandingVatPaymentAuditModel(user: User,
-                                                   payment: Option[Payment]) extends AuditModel {
+                                                   payments: Option[Payments]) extends AuditModel {
 
-  private val paymentDetails: Map[String, String] = payment match {
-    case Some(data) => Map(
+  private val paymentDetails: Map[String, String] = payments match {
+    case Some(data) if data.financialTransactions.size == 1 => Map(
```
Comment (TomDudley92) [inline app/audit/models/ViewNextOutstandingVatPaymentAuditModel.scala]: This if statement isn't needed. If the case below is moved above you can assume if it fails that `if data.financialTransactions.size > 1` then the size will always be 1. 

You can look at the ViewNextOpenVatObligationsAuditModel.scala for an example of this

File: test/services/VatDetailsServiceSpec.scala (hunk 5 above)
Comment (davidmthornton) [reply test/services/VatDetailsServiceSpec.scala]: done

**Pull request merged**